src/neo4j_genai/experimental/components/types.py

```diff
@@ -30,6 +30,7 @@ class TextChunk(BaseModel):
     """
 
     text: str
+    index: int
```

Comment:
Why create an index parameter and not just include it in the metadata?

Reply:
I included it here instead of passing it along with the chunk for most methods of the `ERExtractor`. Which metadata are you talking about here?

Reply:
The `TextChunk` metadata. I think previously we just included the index as a key in the metadata. Now it's its own parameter. Was just wondering why you changed it? It's not a big deal though, makes just as much sense to have it as its own parameter tbh

Reply:
Oh, now I see, sorry. I'd still keep it in a separate property because it is now required and would raise errors in the following components if not there.